src/plugins/xsOverlay/index.tsx

```diff
@@ -136,7 +166,12 @@ const settings = definePluginSettings({
     },
 });
 
-const Native = VencordNative.pluginHelpers.XSOverlay as PluginNative<typeof import("./native")>;
+let socket: WebSocket;
+
+async function start() {
+    if (socket) socket.close();
+    socket = new WebSocket(`ws://127.0.0.1:${settings.store.webSocketPort ?? 42070}/?client=Vencord`);
```

Comment:
it might take some time for the websocket connection to open, which will cause this to fail:
```js
start();
socket.send(...)
```	

You can solve that by making this function return a Promise that resolves once a connection has been opened:
```suggestion
    socket = new WebSocket(`ws://127.0.0.1:${settings.store.webSocketPort ?? 42070}/?client=Vencord`);
    return new Promise<void>((resolve, reject) => {
    	socket.onopen = resolve;
    	socket.onerror = reject;
    	setTimeout(reject, 3000);
    });
```

Now just
```js
await start();
socket.send(...);
```